pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -248,6 +248,28 @@ func (uc *UpgradeCluster) upgradable(currentVersion string) error {
 	return errors.Errorf("%s cannot be upgraded to %s", currentVersion, targetVersion)
 }
 
+// getVMSSNodeVersion returns a VMSS node's current Kubernetes version.
+// If the latest VMSS model is applied, then we can get the version info from the tags.
+// Otherwise, we have to get version via Kubernetes API. This is because VMSS does not support tags
+// for individual instances and old/new instances have the same tags.
+func (uc *UpgradeCluster) getVMSSNodeVersion(client armhelpers.KubernetesClient, name string, tags map[string]*string, latestModelApplied bool) string {
+	if latestModelApplied && tags != nil && tags["orchestrator"] != nil {
+		parts := strings.Split(*tags["orchestrator"], ":")
+		if len(parts) == 2 {
+			return parts[1]
+		}
+	}
+
+	if client != nil {
+		node, err := client.GetNode(name)
+		if err == nil {
+			return strings.TrimPrefix(node.Status.NodeInfo.KubeletVersion, "v")
+		}
+		uc.Logger.Warnf("Failed to get node %s: %v", name, err)
+	}
+	return ""
```

Comment:
See https://github.com/Azure/aks-engine/pull/1299/files#diff-95227b07e97ba7cbd04cfd8b2cdb6980R288

The VMSS case probably needs to either be applied here or removed from the other function if we don't think it's needed anymore since `getNodeVersion` doesn't get used for VMSS anymore.

Also `getVMSSNodeVersion` has a lot of duplicated code with `getNodeVersion` so we can probably reuse some of the code in one common function.

Reply:
Yeap. Thought about doing this but would like to get confirmation first. Removed the VMSS part from getNodeVersions now. Changed the flag name to getVersionFromTags.